pkg/ccl/streamingccl/streamingest/alter_replication_job.go

```diff
@@ -197,33 +179,44 @@ func alterReplicationJobHook(
 
 func alterTenantJobCutover(
 	ctx context.Context,
-	txn isql.Txn,
-	jobRegistry *jobs.Registry,
+	p sql.PlanHookState,
 	ptp protectedts.Storage,
 	alterTenantStmt *tree.AlterTenantReplication,
 	tenInfo *mtinfopb.TenantInfo,
-	cutoverTime hlc.Timestamp,
-) error {
+) (cutoverTime hlc.Timestamp, err error) {
 	if alterTenantStmt == nil || alterTenantStmt.Cutover == nil {
-		return errors.AssertionFailedf("unexpected nil ALTER TENANT cutover expression")
+		return hlc.Timestamp{}, errors.AssertionFailedf("unexpected nil ALTER TENANT cutover expression")
 	}
 
+	jobRegistry := p.ExecCfg().JobRegistry
 	tenantName := tenInfo.Name
-	job, err := jobRegistry.LoadJobWithTxn(ctx, tenInfo.TenantReplicationJobID, txn)
+	job, err := jobRegistry.LoadJobWithTxn(ctx, tenInfo.TenantReplicationJobID, p.InternalSQLTxn())
 	if err != nil {
-		return err
+		return hlc.Timestamp{}, err
 	}
 	details, ok := job.Details().(jobspb.StreamIngestionDetails)
 	if !ok {
-		return errors.Newf("job with id %d is not a stream ingestion job", job.ID())
+		return hlc.Timestamp{}, errors.Newf("job with id %d is not a stream ingestion job", job.ID())
 	}
 	progress := job.Progress()
 	if alterTenantStmt.Cutover.Latest {
 		ts := progress.GetHighWater()
 		if ts == nil || ts.IsEmpty() {
-			return errors.Newf("replicated tenant %q has not yet recorded a safe replication time", tenantName)
+			return hlc.Timestamp{},
+				errors.Newf("replicated tenant %q has not yet recorded a safe replication time", tenantName)
 		}
 		cutoverTime = *ts
+	} else {
+		if alterTenantStmt.Cutover.Timestamp == nil {
+			return hlc.Timestamp{}, errors.AssertionFailedf("unexpected nil cutover expression")
+		}
+
+		evalCtx := &p.ExtendedEvalContext().Context
+		ct, err := evalCutoverTime(ctx, evalCtx, p.SemaCtx(), alterTenantStmt.Cutover.Timestamp)
```

Comment:
We typically try to keep all of these evals outside of the `PlanHookRowFn`. I can't claim to know all the reasons _why_ we need to do that. But perhaps we should keep that pattern here? I know it makes the code a bit uglier.

Alternatively, have I just been cargo-culting this rule around?

Reply:
definitely not cargo-culting anything, thanks Steven for the helpful chat!
PTAL.